src/main/java/bitflow4j/steps/database/WriteToInfluxDB.java

```diff
@@ -0,0 +1,101 @@
+package bitflow4j.steps.database;
+
+import bitflow4j.AbstractPipelineStep;
+import bitflow4j.Sample;
+import bitflow4j.script.registry.Description;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Point;
+import org.influxdb.dto.Pong;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Description("Writes all metrics of samples to the database with the given prefix.")
+public class WriteToInfluxDB extends AbstractPipelineStep {
+
+    private final static int batchModeSize = 100;
+    private final static int batchModeTimeout = 200;
+    private static String databaseName;
+    private final static int reconnectTimeout = 1000;
+
+    private final String databaseURL;
+    private final String username;
+    private final String password;
+    private final String prefix;
+
+    private InfluxDB influxDB;
+
+    protected static final Logger logger = Logger.getLogger(WriteToInfluxDB.class.getName());
+
+    public WriteToInfluxDB(String databaseURL, String databaseName, String username, String password, String prefix) {
+        this.databaseURL = databaseURL;
+        this.username = username;
+        this.password = password;
+        this.prefix = prefix;
+        this.databaseName = databaseName;
+
+        influxDB = InfluxDBFactory.connect(databaseURL, username, password);
+
+        Pong response = influxDB.ping();
+        while (response.getVersion().equalsIgnoreCase("unknown")) {
+            logger.log(Level.SEVERE, String.format("Could not connect to InfluxDB with address '%s'. Retry in %s milliseconds", databaseURL, reconnectTimeout));
```

Comment:
Please add a timeout here, e.g. stop trying after 5 retries. It would also be good to re-establish this connection in writeSample(), in case it has been closed... But this can be left for the future.

Reply:
This is intended behaviour, this service / step shall try to reconnect to the database without stopping. Plus, there is a procedure which reconnects in the writeSample method.